cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/index.js

```diff
@@ -35,31 +37,49 @@ const browserMap = {
 };
 
 export default class DataPrepBrowser extends Component {
-  constructor(props) {
-    super(props);
-    let store = DataPrepBrowserStore.getState();
-    this.state = {
-      activeBrowser: store.activeBrowser
-    };
-  }
-  componentWillMount() {
-    DataPrepBrowserStore.subscribe(() => {
+  state = {
+    activeBrowser: DataPrepBrowserStore.getState().activeBrowser
+  };
+
+  componentDidMount() {
+    this.storeSubscription = DataPrepBrowserStore.subscribe(() => {
       let {activeBrowser} = DataPrepBrowserStore.getState();
       if (activeBrowser.name && this.state.activeBrowser.name !== activeBrowser.name) {
         this.setState({
           activeBrowser
         });
       }
     });
+    if (typeof this.props.setActiveBrowser === 'function') {
+      this.props.setActiveBrowser();
+    }
+  }
+
+  componentDidUpdate() {
+    if (typeof this.props.setActiveBrowser === 'function') {
+      this.props.setActiveBrowser();
```

Comment:
This would not be any different than calling this function in the render prop of `Link` does it? This will get invoked whenever the user is interacting within, say the GCS browser right?

I don't think calling this in CDU would be efficient for us.